linter.py

```diff
@@ -0,0 +1,39 @@
+import os
+import re
+from typing import Pattern, List
+
+search_note_pattern: Pattern = re.compile(r'^\*\*Note\*\*')
+search_warning_pattern: Pattern = re.compile(r'^\*\*Warning\*\*')
+correct_note_pattern: Pattern = re.compile(r'^> \*\*Note\*\*')
+correct_warning_pattern: Pattern = re.compile(r'^> \*\*Warning\*\*')
+
+def find_regex_in_file_content(
+    search_pattern: Pattern,
+    correct_pattern: Pattern,
+    filenames: List[str]
+) -> bool:
+    """Function that looks for a regex pattern in the given file contents."""
+
+    exit_code = 0
+    for filename in filenames:
+        with open(filename) as file:
+            lines = file.readlines()
+        for i, line in enumerate(lines):
```

Comment:
You can also just do `for i, line in enumerate(file)`, which is slightly more memory-efficient because you only load one line at a time in to memory instead of loading the whole file.

Reply:
updated